src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -969,6 +976,8 @@ private void CreateResponseHeader(bool appCompleted)
                 responseHeaders.SetRawDate(dateHeaderValues.String, dateHeaderValues.Bytes);
             }
 
+            _keepAlive = keepAlive;
```

Comment:
Only if `appCompleted == false` and `RejectNonBodyTransferEncodingResponse` is called. Should `_keepAlive` be set in that path? (Also should it always be false on error?)